odbc/src/tests.rs

```diff
@@ -287,3 +291,163 @@ fn invalid_alloc() {
         );
     }
 }
+
+fn validate_diag_rec(handle_type: HandleType, handle: *mut MongoHandle) {
```

Comment:
if this function is only used as a helper within the `simple_diag_rec` test, let's define it in that function instead of at the module level